text/0000-more-api-documentation-conventions.md

```diff
@@ -0,0 +1,726 @@
+
+Other places do not know how to highlight this anyway, so it's not important to
+be explicit.
+
+RFC 505 suggests that references and citation should be linked ‘reference
+style.’ This is still recommended, but prefer to leave off the second `[]`:
+
+```
+[Rust website]
+
+[Rust website]: http://www.rust-lang.org
+```
+
+to
+
+```
+[Rust website][website]
+
+[website]: http://www.rust-lang.org
+```
+
+But, if the text is very long, it is okay to use this form.
+
+### Examples in API docs
+[examples-in-api-docs]: #examples-in-api-docs
+
+Everything should have examples. Here is an example of how to do examples:
+
+```
+/// # Examples
+///
+/// Basic usage:
+///
+/// ```
+/// use op;
+///
+/// let s = "foo";
+/// let answer = op::compare(s, "bar");
+/// ```
+///
+/// Passing a closure to compare with, rather than a string:
+///
+/// ```
+/// use op;
+///
+/// let s = "foo";
+/// let answer = op::compare(s, |a| a.chars().is_whitespace().all());
+/// ```
+```
+
+For particularly simple APIs, still say “Examples” and “Basic usage:” for
+consistency’s sake.
+
+### Referring to types
+[referring-to-types]: #referring-to-types
+
+When talking about a type, use its full name. In other words, if the type is generic,
+say `Option<T>`, not `Option`. An exception to this is bounds. Write `Cow<'a, B>`
+rather than `Cow<'a, B> where B: 'a + ToOwned + ?Sized`.
+
+Another possibility is to write in lower case using a more generic term. In other words,
+‘string’ can refer to a `String` or an `&str`, and ‘an option’ can be ‘an `Option<T>`’.
+
+### Use parentheses for functions
+[use-parentheses-for-functions]: #use-parentheses-for-functions
+
+When referring to function names, include the `()`s after the name. For example, do this:
+
+```rust
+/// This behavior is similar to the way that `mem::replace()` works.
+```
+
+Not this:
+
+```rust
+/// This behavior is similar to the way that `mem::replace` works.
+```
+
+This helps visually differentiate it in the text.
+
+### Link all the things
+[link-all-the-things]: #link-all-the-things
+
+A major drawback of Markdown is that it cannot automatically link types in API documentation.
+Do this yourself with the reference-style syntax, for ease of reading:
+
+```
+/// The [`String`] passed in lorum ipsum...
+///
+/// [`String`]: ../string/struct.String.html
+```
+
+### Module-level vs type-level docs
+[module-level-vs-type-level-docs]: #module-level-vs-type-level-docs
+
+There has often been a tension between module-level and type-level
+documentation. For example, in today's standard library, the various
+`*Cell` docs say, in the pages for each type, to "refer to the module-level
+documentation for more details."
+
+Instead, module-level documentation should show a high-level summary of
+everything in the module, and each type should document itself fully. It is
+okay if there is some small amount of duplication here. Module-level
+documentation should be broad, and not go into a lot of detail, which is left
```

Comment:
And I also think the first comma in the sentence is too much:

"documentation should be broad and not go into a lot of detail"